### PR TITLE
Add support for machine jetson-xavier-seeed-hat

### PR DESCRIPTION
**Pull request by KunalAivero**
1. add new machine type jetson-xavier-seeed-hat
2. base files based on jetson-xavier-seeed-hat with modified dtb
3. based on L4T version 32.4.4

### CODE REVIEW

**Comment by alexgg** [comment]
@resin-jenkins ok to test

**Comment by alexgg** [comment]
@balena-ci rebase

**Comment by KunalAivero** [comment]
@alexgg I have rebased to master

File: jetson-xavier-seeed-hat.coffee [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{ networkOptions, commonImg, instructions } = deviceTypesCommon
+
+BOARD_POWEROFF = 'Remove power from the Jetson Xavier NX Devkit SD-CARD and insert the freshly burnt sd-card.'
+BOARD_POWERON  = 'Power on the board.'
+
+module.exports =
+	version: 1
+	slug: 'jetson-xavier-seeed-hat'
+	aliases: [ 'jetson-xavier-seeed-hat' ]
+	name: 'Nvidia Jetson Xavier NX Devkit with seeed HAT'
```
Comment (acostach) [inline jetson-xavier-seeed-hat.coffee:11]: Also suggesting a more descriptive name like "Nvidia Jetson Xavier NX Devkit SD-CARD <Seeed HAT name>" or, if it has a short product code, even better.

File: jetson-xavier-seeed-hat.coffee [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	arch: 'aarch64'
+	state: 'new'
+
```
Comment (acostach) [inline jetson-xavier-seeed-hat.coffee:14]: Let's also mark this as a community board maintained by you. This is done by adding "community: true" here.

File: jetson-xavier-seeed-hat.coffee [hunk 3]
```diff
@@ -0,0 +1,44 @@
+deviceTypesCommon = require '@resin.io/device-types/common'
+{ networkOptions, commonImg, instructions } = deviceTypesCommon
+
+BOARD_POWEROFF = 'Remove power from the Jetson Xavier NX Devkit SD-CARD and insert the freshly burnt sd-card.'
+BOARD_POWERON  = 'Power on the board.'
+
+module.exports =
+	version: 1
+	slug: 'jetson-xavier-seeed-hat'
```
Comment (acostach) [inline jetson-xavier-seeed-hat.coffee:9]: I suggest we include the keywords 'nx' and 'sd' in the slug, yocto machine and coffee file name so it's more clear that this is not a Xavier AGX.

File: jetson-xavier-seeed-hat.coffee [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+	gettingStartedLink:
+		windows: 'https://docs.balena.io/jetson-xavier-nx-devkit-sd-card/nodejs/getting-started/#adding-your-first-device'
+		osx: 'https://docs.balena.io/jetson-xavier-nx-devkit-sd-card/nodejs/getting-started/#adding-your-first-device'
+		linux: 'https://docs.balena.io/jetson-xavier-nx-devkit-sd-card/nodejs/getting-started/#adding-your-first-device'
+
+	supportsBlink: false
+
+	yocto:
+		machine: 'jetson-xavier-seeed-hat'
```
Comment (acostach) [inline jetson-xavier-seeed-hat.coffee:29]: Same here

**Review by acostach — COMMENTED** [review]
Thank you for your PR, I left a couple comments, perhaps we can find a more descriptive device name for the board. Otherwise it looks fine. I assume you tested a local build with your board and the hat worked as expected, right?

**Comment by KunalAivero** [comment]
@acostach I implemented the review comments. However, i have not tested the jetson-xavier-nx-devkit-seeed-hat images as I can't build the images without having valid device-id for this board. The issue is [discussed here on balena forum](https://forums.balena.io/t/seeking-a-new-machine-or-target-for-our-hardware/323062/10). Can you please provide your feedback on this issue.

**Comment by acostach** [comment]
@KunalAivero does it work if you pass --type jetson-xavier-nx-devkit to the inject command? Also, you might need to mount the boot partition and edit device-type.json to replace the slug to 'jetson-xavier-nx-devkit' too for the purpose of this test, before writing the image to the device. To check if it boots properly you connect the device to ethernet or wifi and use 'balena scan'.

**Comment by KunalAivero** [comment]
@acostach no it does not work if I try to inject the config with machine-id "jetson-xavier-nx-devkit"
```
sudo ~/tools-balena/balena-cli/balena config inject ~/balena_workspace/BuildBalena/xavier_nx_config_files/staging-aivero-do-linux-armv8-l4t-t186.config.json --drive ./build/tmp/deploy/images/jetson-xavier-nx-devkit-seeed-hat/balena-image-jetson-xavier-nx-devkit-seeed-hat.balenaos-img  --type jetson-xavier-nx-devkit
[sudo] password for xyz: 
Partition not found: 9.
```
I need to figure out why I get the error as above. Do you have any quick suggestions regarding what I can check.

**Comment by acostach** [comment]
You can check the image you built with parted -s <image>.balenaos.img unit s print or with fdisk to see if the partitions look right. The resin-boot partition should be the 9th on the sd-card, after the nvidia partitions: https://github.com/balena-os/balena-jetson/blob/master/layers/meta-balena-jetson/recipes-bsp/tegra-binaries/files/partition_specification194_nxde_sdcard.txt

**Comment by KunalAivero** [comment]
@acostach Thanks! that indeed was the problem.
(1) After I made this fix (see the last commit) the sdcard image is good
(2) I can inject config in to sdcard img file using --type jetson-xavier-nx-devkit
(3) the balena scan find this new device
```
->sudo tools-balena/balena-cli/balena scan
[sudo] password for xyz: 
Scanning for local balenaOS devices... Reporting scan results
- 
  host:      d714ff5.local
  address:   192.168.1.17
  osVariant: production
```

**Comment by acostach** [comment]
Great to hear that @KunalAivero , did you manage to find a product code for the HAT so that it's more obvious for the users which product is being used? Seeed manufactures many hats so a short code or name for the HAT in the coffee file at "name: " would help. 

Also, please note that each commit needs to have a "Signed-off-by: You <your@email>" and only one of the commits will need to also have a "Changelog-entry", please see the other PRs for example. This is needed for the github checks to pass. Thanks

**Comment by acostach** [comment]
@resin-jenkins test this please

**Review by acostach — APPROVED** [review]

**Comment by compiaffe** [comment]
@acostach I see this type now shows up on https://dashboard.balena-staging.com but when I try to add a device the left part of the interface does not load. I briefly see `Fetching Version` and then it stays as such:

image

Is there still something missing to deploy devices from the dashboard?

**Comment by acostach** [comment]
Hi @compiaffe, this device has the device-type application created but does not have an image available just yet, as the staging environment is only meant for testing. Will let you know once the OS image becomes available.

**Comment by compiaffe** [comment]
@acostach I just built it locally and tried to inject a config using the balena-cli 12.44.20, which complains about an `Invalid device type: jetson-xavier-nx-devkit-seeed-2mic-hat` - Is there a branch on the cli which already works and I can test?

**Comment by acostach** [comment]
@compiaffe the staging build is now available for download, can you please check again?

**Comment by compiaffe** [comment]
@acostach I can confirm it is there now. Thank you :)

**Comment by acostach** [comment]
Hi @KunalAivero, we've relased the Xavier AGX at 2.82.11 with L4T 32.5.1, please update the device tree with one based on 32.5.1 for the respeaker device-type and we'll release an image for it too.

**Comment by KunalAivero** [comment]
@acostach  I have created a new [MR-208](https://github.com/balena-os/balena-jetson/pull/208) for device-tree based on L4T 32.5.1

**Pull request merged**